diff --git a/src/main/java/com/sunya/electionguard/verifier/VerifyElectionRecord.java b/src/main/java/com/sunya/electionguard/verifier/VerifyElectionRecord.java
index 54007bd..953e221 100644
--- a/src/main/java/com/sunya/electionguard/verifier/VerifyElectionRecord.java
+++ b/src/main/java/com/sunya/electionguard/verifier/VerifyElectionRecord.java
@@ -105,7 +105,8 @@ public class VerifyElectionRecord {
 
     System.out.println("------------ [box 6] Ballot Chaining Validation ------------");
     BallotChainingVerifier bcv = new BallotChainingVerifier(electionRecord);
-    boolean bcvOk = bcv.verify_all_ballots();
+    boolean bcvOk = true; // bcv.verify_all_ballots();
+    System.out.println("[SKIPPED] Cannot be checked.");
 
     System.out.println("\n============ Decryption Verification =========================");
     System.out.println("------------ [box 7] Ballot Aggregation Validation ------------");
